packages/block-editor/src/components/block-list/use-block-props/use-focus-first-element.js

```diff
@@ -101,7 +101,13 @@ export function useFocusFirstElement( clientId ) {
 		const target =
 			( isReverse ? last : first )( textInputs ) || ref.current;
 
-		if ( ! isInsideRootBlock( ref.current, target ) ) {
+		// Backspacing into a caption field should select the parent block
+		// see https://github.com/WordPress/gutenberg/issues/7422
+		// This prevents situation where the caption 'hijacks' focus
+		// and the caret is stuck inside the caption input.
+		const targetIsCaption = target?.nodeName === 'FIGCAPTION';
+
+		if ( targetIsCaption || ! isInsideRootBlock( ref.current, target ) ) {
```

Comment:
For this code to work, it likely needs to be placed after my code on this line.

https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-list/use-block-props/use-focus-first-element.js#L115

If not, it will likely just add focus back to the caption field.

I am not sure this is the best way to handle this either. Seems like putting focus on the last element traveling backwards is the right thing to do.

Reply:
In that case , do you think if we go with @tellthemachines 's suggestion in this [comment](https://github.com/WordPress/gutenberg/pull/39484#pullrequestreview-921151425) , would that be more in line with putting focus on the last element traveling backwards